apps/theming/lib/Controller/ThemingController.php

```diff
@@ -203,8 +203,8 @@ public function updateStylesheet($setting, $value) {
 	 * Check that a string is a valid http/https url
 	 */
 	private function isValidUrl(string $url): bool {
-		return ((strpos($url, 'http://') === 0 || strpos($url, 'https://') === 0) &&
-			filter_var($url, FILTER_VALIDATE_URL) !== false);
+		// Regex copied from https://stackoverflow.com/a/206087 but without ftp support
+		return preg_match('#(https?://(\S*?\.\S*?|localhost))([\s)\[\]{},;"\':<]|\.\s|$)#i', $url) === 1;
```

Comment:
I'd suggest to adjust the regex to also allow hostnames without the domain part other than localhost only

```suggestion
		return preg_match('#(https?://(\S*?\.*\S*?))([\s)\[\]{},;"\':<]|\.\s|$)#i', $url) === 1;
```

Reply:
Or just to safeguard, we could only use the regex as a fallback if the filter_var doesn't match:

```suggestion
		return filter_var($url, FILTER_VALIDATE_URL) !== false) || preg_match('#(https?://(\S*?\.\S*?|localhost))([\s)\[\]{},;"\':<]|\.\s|$)#i', $url) === 1;
```

Reply:
It turns out there is a idn_to_ascii function that makes it possible to then use filter_var. Fount it in https://github.com/nextcloud/server/issues/27546.